Library/Homebrew/formulary.rb

```diff
@@ -251,4 +253,29 @@ def self.tap_paths(name)
       ]).detect(&:file?)
     end.compact
   end
+
+  def self.tap_name(path)
+    match = path =~ HOMEBREW_TAP_PATH_REGEX
+    match ? "#{$1}/#{$2.sub("homebrew-", "")}" : "homebrew/homebrew"
+  end
+
+  def self.pinned_tap_paths(name)
+    tap_paths = self.tap_paths(name)
+    tap_paths.select { |x| Tap.in_pinned_tap?(x.to_s) }
+  end
```

Comment:
We can avoid this method and `Tap.in_pinned_tap?`( therefore to avoid some unnecessary string regex matching). To do that, we can change `Fomulary.tap_paths` to:
```ruby
  def self.tap_paths(name, taps=Dir["#{HOMEBREW_LIBRARY}/Taps/*/*/"])
    name = name.downcase
    taps.map do |tap|
      Pathname.glob([
        "#{tap}Formula/#{name}.rb",
        "#{tap}HomebrewFormula/#{name}.rb",
        "#{tap}#{name}.rb",
      ]).detect(&:file?)
    end.compact
  end
```

So in `find_with_priority`, we can just do
```ruby
possible_pinned_tap_formulae = tap_paths(ref, Taps.select(&:pinned).map(&:path))
```